tests/testthat/test-cube-mr.R

```diff
@@ -102,26 +102,97 @@ test_that("prop.table(cell) for as_selected x cat", {
     expect_equal(as_selected_margins(2, nothing_selected), 2)
     expect_equal(as_selected_margins(NULL, nothing_selected), NULL)
+    selecteds <- c(FALSE, TRUE, FALSE) # MR by cat
+    expect_equal(as_selected_margins(1, selecteds), 1)
+    expect_equal(as_selected_margins(2, selecteds), c(1, 3))
+    expect_equal(as_selected_margins(NULL, selecteds), 1)
+    selecteds <- c(FALSE, FALSE, TRUE) # cat by MR
+    expect_equal(as_selected_margins(1, selecteds), c(1, 2))
+    expect_equal(as_selected_margins(2, selecteds), 2)
+    expect_equal(as_selected_margins(NULL, selecteds), 2)
+    selecteds <- c(FALSE, FALSE, TRUE, FALSE, TRUE) # cat by MR by MR
+    expect_equal(as_selected_margins(1, selecteds), c(1, 2, 4))
+    expect_equal(as_selected_margins(2, selecteds), c(2, 4))
+    expect_equal(as_selected_margins(3, selecteds), c(2, 4))
+    expect_equal(as_selected_margins(NULL, selecteds), c(2, 4))
+    selecteds <- c(FALSE, FALSE, TRUE, FALSE, TRUE, FALSE, TRUE) # cat by MR*3
+    expect_equal(as_selected_margins(1, selecteds), c(1, 2, 4, 6))
+    expect_equal(as_selected_margins(2, selecteds), c(2, 4, 6))
+    expect_equal(as_selected_margins(3, selecteds), c(2, 4, 6))
+    expect_equal(as_selected_margins(4, selecteds), c(2, 4, 6))
+    expect_equal(as_selected_margins(NULL, selecteds), c(2, 4, 6))
 })
 
 test_that("as_selected_margins with before=FALSE", {
-    selecteds <- c(FALSE, TRUE, FALSE)
-    expect_equal(as_selected_margins(1, selecteds, before=FALSE), 1)
-    expect_equal(as_selected_margins(2, selecteds, before=FALSE), c(1, 2))
-    expect_equal(as_selected_margins(NULL, selecteds, before=FALSE), 1)
-    nothing_selected <- c(FALSE, FALSE)
+    nothing_selected <- c(FALSE, FALSE)# cat by cat
     expect_equal(as_selected_margins(1, nothing_selected, before=FALSE), 1)
     expect_equal(as_selected_margins(2, nothing_selected, before=FALSE), 2)
     expect_equal(as_selected_margins(NULL, nothing_selected, before=FALSE), NULL)
+    selecteds <- c(FALSE, TRUE, FALSE) # MR by cat
+    expect_equal(as_selected_margins(1, selecteds, before=FALSE), 1)
+    expect_equal(as_selected_margins(2, selecteds, before=FALSE), c(1, 2))
+    expect_equal(as_selected_margins(NULL, selecteds, before=FALSE), 1)
+    selecteds <- c(FALSE, FALSE, TRUE) # cat by MR
+    expect_equal(as_selected_margins(1, selecteds, before=FALSE), c(1, 2))
+    expect_equal(as_selected_margins(2, selecteds, before=FALSE), 2)
+    expect_equal(as_selected_margins(NULL, selecteds, before=FALSE), 2)
+    selecteds <- c(FALSE, FALSE, TRUE, FALSE, TRUE) # cat by MR by MR
+    expect_equal(as_selected_margins(1, selecteds, before=FALSE), c(1, 2, 3))
+    expect_equal(as_selected_margins(2, selecteds, before=FALSE), c(2, 3))
+    expect_equal(as_selected_margins(3, selecteds, before=FALSE), c(2, 3))
+    expect_equal(as_selected_margins(NULL, selecteds, before=FALSE), c(2, 3))
+    selecteds <- c(FALSE, FALSE, TRUE, FALSE, TRUE, FALSE, TRUE) # cat by MR*3
```

Comment:
Try something like MR x cat x MR, just to be sure that we aren't assuming anything about where the MRs are.